compiler/resolution/functionResolution.cpp

```diff
@@ -6457,7 +6553,46 @@ resolveExpr(Expr* expr) {
         implementForallIntents2wrapper(call, eflopiHelper);
       }
 
-      resolveFns(call->isResolved());
+      // For ContextCallExprs, be sure to resolve all of the
+      // functions that could be called.
+      if (ContextCallExpr* cc = toContextCallExpr(call->parentExpr)) {
+        Type* firstRetType = NULL;
+        Type* retType = NULL;
+        bool same = true;
+        for_alist(optionExpr, cc->options) {
+          CallExpr* callOption = toCallExpr(optionExpr);
+          FnSymbol* calledFn = callOption->isResolved();
+          INT_ASSERT(calledFn);
+          resolveFns(calledFn);
+          INT_ASSERT(calledFn->retType);
```

Comment:
This assertion will never fail, as retType is at least _unknown. A better assert might be retType != dtUnknown && retType != dtAny.